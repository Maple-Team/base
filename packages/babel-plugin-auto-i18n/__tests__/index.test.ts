import { getTransformCode } from './test-help'

describe('auto i18n plugin unit tests', () => {
  it('handle JSXText', () => {
    const sourceCode = `
    export const Component = () => {
      return <div>你好</div>
    }
    `
    const result = getTransformCode(sourceCode, 'jsxtext.tsx')
    expect(result?.code).toMatchSnapshot()
  })

  describe('stringLiteral cases', () => {
    it('handle console', () => {
      const sourceCode = `
      export const Component = () => {
        console.log('输出值')
        return <div name="属性">jsx文本</div>
      }
      `
      const result = getTransformCode(sourceCode, 'jsxattribute.tsx')
      expect(result?.code).toMatchSnapshot()
    })

    it('handle JSXAttribute', () => {
      const sourceCode = `
      export const Component = () => {
        return <div name="属性">jsx文本</div>
      }
      `
      const result = getTransformCode(sourceCode, 'jsxattribute.tsx')
      expect(result?.code).toMatchSnapshot()
    })

    it('handle Assignment', () => {
      const sourceCode = `
      export const Component = () => {
        const var1:string = "变量1"
        return <div name="属性">站位文本</div>
      }
      `
      const result = getTransformCode(sourceCode, 'assignment.tsx')
      expect(result?.code).toMatchSnapshot()
    })
    it('handle objectProperty', () => {
      const sourceCode = `
      export const Component = () => {
        const obj1 = {
          name: "属性值1",
          key: "属性值2"
        }
        return <div name="属性">站位文本</div>
      }
      `
      const result = getTransformCode(sourceCode, 'objectProperty.tsx')
      expect(result?.code).toMatchSnapshot()
    })
    it('handle module scope objectProperty', () => {
      const sourceCode = `
      const obj1 = {
        name: "属性值1",
        key: "属性值2"
      }
      export const Component = () => {
        return <div name="属性">站位文本</div>
      }
      `
      const result = getTransformCode(sourceCode, 'objectProperty-module.tsx')
      expect(result?.code).toMatchSnapshot()
    })
    it('handle arrayExpression', () => {
      const sourceCode = `
      export const Component = () => {
        const obj1 = ["属性值1", "属性值2"]
        return <div name="属性">站位文本</div>
      }
      `
      const result = getTransformCode(sourceCode, 'arrayExpression.tsx')
      expect(result?.code).toMatchSnapshot()
    })
    it('handle svg react component', () => {
      const sourceCode = `
      export const Component = () => {
      return <svg
        t="1711425339799"
        className="icon"
        viewBox="0 0 1024 1024"
        version="1.1"

        xmlns="http://www.w3.org/2000/svg"
        width="200"
        height="200"
      >
        <path
        id="svg中的中文"
          d="M448 672c-6.4 0-19.2 0-25.6-6.4-12.8-12.8-12.8-32 0-44.8L531.2 512 422.4 409.6c-12.8-12.8-12.8-32 0-44.8s32-12.8 44.8 0l128 128c12.8 12.8 12.8 32 0 44.8l-128 128C467.2 672 454.4 672 448 672z"
          fill="#272636"
        />
      </svg>
      }
      `
      const result = getTransformCode(sourceCode, 'arrayExpression.tsx')
      expect(result?.code).toMatchSnapshot()
    })
  })
  describe('handle useEffect hooks add t as dependences', () => {
    it('exported function case 0', () => {
      const sourceCode = `
      export const Component = () => {
        const [v, setV] = useState<string>('1')
        useEffect(()=>{
          const text = "effect作用域中的文案1"
          console.log(v)
        },[v])
        return <div name="属性">站位文本</div>
      }
      `
      const result = getTransformCode(sourceCode, 'useEffect-exported-function0.tsx')
      expect(result?.code).toMatchSnapshot()
    })
    it('exported function case 1', () => {
      const sourceCode = `
      export const Component = () => {
        useEffect(()=>{
          const text = "effect作用域中的文案1"
        },[])
        return <div name="属性">站位文本</div>
      }
      `
      const result = getTransformCode(sourceCode, 'useEffect-exported-function1.tsx')
      expect(result?.code).toMatchSnapshot()
    })
    it('exported function case 2', () => {
      const sourceCode = `
      export const Component = () => {
        useEffect(()=>{
          const text = "effect作用域中的文案2"
        })
        return <div name="属性">站位文本</div>
      }
      `
      const result = getTransformCode(sourceCode, 'useEffect-exported-function2.tsx')
      expect(result?.code).toMatchSnapshot()
    })
  })
  describe('handle useCallback hooks add t as dependences', () => {
    it('exported function case 0', () => {
      const sourceCode = `
      export const Component = () => {
        const [v, setV] = useState<string>('1')
        const cb1 = useCallback(()=>{
          const text = "useCallback作用域中的文案0"
          console.log(v)
        },[v])
        return <div name="属性" onClick={cb1}>站位文本</div>
      }
      `
      const result = getTransformCode(sourceCode, 'useCallback-exported-function0.tsx')
      expect(result?.code).toMatchSnapshot()
    })
    it('exported function case 1', () => {
      const sourceCode = `
      export const Component = () => {
        const cb1 = useCallback(()=>{
          const text = "useCallback作用域中的文案1"
        },[])
        return <div name="属性" onClick={cb1}>站位文本</div>
      }
      `
      const result = getTransformCode(sourceCode, 'useCallback-exported-function1.tsx')
      expect(result?.code).toMatchSnapshot()
    })
  })
  describe('handle useMemo hooks add t as dependences', () => {
    it('exported function case 0', () => {
      const sourceCode = `
      export const Component = () => {
        const cb1 = useMemo(()=>{
          const text = "useMemo作用域中的文案1"
          return () => {
            console.log(text)
          }
        },[])
        return <div name="属性" onClick={cb1}>站位文本</div>
      }
      `
      const result = getTransformCode(sourceCode, 'useMemo-exported-function0.tsx')
      expect(result?.code).toMatchSnapshot()
    })
    it('exported function case 1', () => {
      const sourceCode = `
      export const Component = () => {
        const text = useMemo(()=>{
          const text = "useMemo作用域中的文案1"
          return text
        },[])
        return <div name="属性">{text}</div>
      }
      `
      const result = getTransformCode(sourceCode, 'useMemo-exported-function1.tsx')
      expect(result?.code).toMatchSnapshot()
    })
    it('exported function case 2', () => {
      const sourceCode = `
      export const Component = () => {
        const var1 = "外部变量1"
        const var2 = useMemo(()=>{
          const text = "text" + var1
          return text
        },[var1])
        return <div name="属性">{var2}</div>
      }
      `
      const result = getTransformCode(sourceCode, 'useMemo-exported-function2.tsx')
      expect(result?.code).toMatchSnapshot()
    })
    it('exported function case 3', () => {
      const sourceCode = `
      export const Component = () => {
        const data = useMemo(()=>[
          {
            name:"值1",
            label: "标签1"
          },
          {
            name:"值2",
            label: "标签2"
          },
          {
            name:"值3",
            label: "标签3"
          }
        ],[])
        return <CustomComponent name="属性" data={data}></CustomComponent>
      }
      `
      const result = getTransformCode(sourceCode, 'useMemo-exported-function2.tsx')
      expect(result?.code).toMatchSnapshot()
    })
  })
})
