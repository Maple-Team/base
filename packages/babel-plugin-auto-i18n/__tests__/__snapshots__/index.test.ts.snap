// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`auto i18n plugin unit tests handle JSXText 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.Component = void 0;
var _reactI18next = require(\\"react-i18next\\");
var Component = exports.Component = function Component() {
  var _useTranslation = (0, _reactI18next.useTranslation)(),
    t = _useTranslation.t,
    i18n = _useTranslation.i18n;
  return <div>{t(\\"440ee085\\")}</div>;
};"
`;

exports[`auto i18n plugin unit tests handle useCallback hooks add t as dependences exported function case 0 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.Component = void 0;
var _reactI18next = require(\\"react-i18next\\");
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError(\\"Invalid attempt to destructure non-iterable instance.\\\\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.\\"); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === \\"string\\") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === \\"Object\\" && o.constructor) n = o.constructor.name; if (n === \\"Map\\" || n === \\"Set\\") return Array.from(o); if (n === \\"Arguments\\" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : \\"undefined\\" != typeof Symbol && r[Symbol.iterator] || r[\\"@@iterator\\"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t[\\"return\\"] && (u = t[\\"return\\"](), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
var Component = exports.Component = function Component() {
  var _useTranslation = (0, _reactI18next.useTranslation)(),
    t = _useTranslation.t,
    i18n = _useTranslation.i18n;
  var _useState = useState('1'),
    _useState2 = _slicedToArray(_useState, 2),
    v = _useState2[0],
    setV = _useState2[1];
  var cb1 = useCallback(function () {
    var text = t(\\"c056fb8f\\");
    console.log(v);
  }, [v, t]);
  return <div name={t(\\"7d3c35c2\\")} onClick={cb1}>{t(\\"e2fd33fa\\")}</div>;
};"
`;

exports[`auto i18n plugin unit tests handle useCallback hooks add t as dependences exported function case 1 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.Component = void 0;
var _reactI18next = require(\\"react-i18next\\");
var Component = exports.Component = function Component() {
  var _useTranslation = (0, _reactI18next.useTranslation)(),
    t = _useTranslation.t,
    i18n = _useTranslation.i18n;
  var cb1 = useCallback(function () {
    var text = t(\\"6a7ba358\\");
  }, [t]);
  return <div name={t(\\"7d3c35c2\\")} onClick={cb1}>{t(\\"e2fd33fa\\")}</div>;
};"
`;

exports[`auto i18n plugin unit tests handle useEffect hooks add t as dependences exported function case 0 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.Component = void 0;
var _reactI18next = require(\\"react-i18next\\");
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError(\\"Invalid attempt to destructure non-iterable instance.\\\\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.\\"); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === \\"string\\") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === \\"Object\\" && o.constructor) n = o.constructor.name; if (n === \\"Map\\" || n === \\"Set\\") return Array.from(o); if (n === \\"Arguments\\" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : \\"undefined\\" != typeof Symbol && r[Symbol.iterator] || r[\\"@@iterator\\"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t[\\"return\\"] && (u = t[\\"return\\"](), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
var Component = exports.Component = function Component() {
  var _useTranslation = (0, _reactI18next.useTranslation)(),
    t = _useTranslation.t,
    i18n = _useTranslation.i18n;
  var _useState = useState('1'),
    _useState2 = _slicedToArray(_useState, 2),
    v = _useState2[0],
    setV = _useState2[1];
  useEffect(function () {
    var text = t(\\"b3105269\\");
    console.log(v);
  }, [v, t]);
  return <div name={t(\\"7d3c35c2\\")}>{t(\\"e2fd33fa\\")}</div>;
};"
`;

exports[`auto i18n plugin unit tests handle useEffect hooks add t as dependences exported function case 1 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.Component = void 0;
var _reactI18next = require(\\"react-i18next\\");
var Component = exports.Component = function Component() {
  var _useTranslation = (0, _reactI18next.useTranslation)(),
    t = _useTranslation.t,
    i18n = _useTranslation.i18n;
  useEffect(function () {
    var text = t(\\"b3105269\\");
  }, [t]);
  return <div name={t(\\"7d3c35c2\\")}>{t(\\"e2fd33fa\\")}</div>;
};"
`;

exports[`auto i18n plugin unit tests handle useEffect hooks add t as dependences exported function case 2 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.Component = void 0;
var _reactI18next = require(\\"react-i18next\\");
var Component = exports.Component = function Component() {
  var _useTranslation = (0, _reactI18next.useTranslation)(),
    t = _useTranslation.t,
    i18n = _useTranslation.i18n;
  useEffect(function () {
    var text = t(\\"0bda56d6\\");
  });
  return <div name={t(\\"7d3c35c2\\")}>{t(\\"e2fd33fa\\")}</div>;
};"
`;

exports[`auto i18n plugin unit tests handle useMemo hooks add t as dependences exported function case 0 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.Component = void 0;
var _reactI18next = require(\\"react-i18next\\");
var Component = exports.Component = function Component() {
  var _useTranslation = (0, _reactI18next.useTranslation)(),
    t = _useTranslation.t,
    i18n = _useTranslation.i18n;
  var cb1 = useMemo(function () {
    var text = t(\\"d1e010b9\\");
    return function () {
      console.log(text);
    };
  }, [t]);
  return <div name={t(\\"7d3c35c2\\")} onClick={cb1}>{t(\\"e2fd33fa\\")}</div>;
};"
`;

exports[`auto i18n plugin unit tests handle useMemo hooks add t as dependences exported function case 1 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.Component = void 0;
var _reactI18next = require(\\"react-i18next\\");
var Component = exports.Component = function Component() {
  var _useTranslation = (0, _reactI18next.useTranslation)(),
    t = _useTranslation.t,
    i18n = _useTranslation.i18n;
  var text = useMemo(function () {
    var text = t(\\"d1e010b9\\");
    return text;
  }, [t]);
  return <div name={t(\\"7d3c35c2\\")}>{text}</div>;
};"
`;

exports[`auto i18n plugin unit tests handle useMemo hooks add t as dependences exported function case 2 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.Component = void 0;
var _reactI18next = require(\\"react-i18next\\");
var Component = exports.Component = function Component() {
  var _useTranslation = (0, _reactI18next.useTranslation)(),
    t = _useTranslation.t,
    i18n = _useTranslation.i18n;
  var var1 = t(\\"94fb4488\\");
  var var2 = useMemo(function () {
    var text = \\"text\\" + var1;
    return text;
  }, [var1]);
  return <div name={t(\\"7d3c35c2\\")}>{var2}</div>;
};"
`;

exports[`auto i18n plugin unit tests handle useMemo hooks add t as dependences exported function case 3 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.Component = void 0;
var _reactI18next = require(\\"react-i18next\\");
var Component = exports.Component = function Component() {
  var _useTranslation = (0, _reactI18next.useTranslation)(),
    t = _useTranslation.t,
    i18n = _useTranslation.i18n;
  var data = useMemo(function () {
    return [{
      name: t(\\"39899bf2\\"),
      label: t(\\"87d45b4f\\")
    }, {
      name: t(\\"1af30f22\\"),
      label: t(\\"e5850bd3\\")
    }, {
      name: t(\\"fa6f1284\\"),
      label: t(\\"244347a9\\")
    }];
  }, [t]);
  return <CustomComponent name={t(\\"7d3c35c2\\")} data={data}></CustomComponent>;
};"
`;

exports[`auto i18n plugin unit tests stringLiteral cases handle Assignment 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.Component = void 0;
var _reactI18next = require(\\"react-i18next\\");
var Component = exports.Component = function Component() {
  var _useTranslation = (0, _reactI18next.useTranslation)(),
    t = _useTranslation.t,
    i18n = _useTranslation.i18n;
  var var1 = t(\\"9e898cce\\");
  return <div name={t(\\"7d3c35c2\\")}>{t(\\"e2fd33fa\\")}</div>;
};"
`;

exports[`auto i18n plugin unit tests stringLiteral cases handle JSXAttribute 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.Component = void 0;
var _reactI18next = require(\\"react-i18next\\");
var Component = exports.Component = function Component() {
  var _useTranslation = (0, _reactI18next.useTranslation)(),
    t = _useTranslation.t,
    i18n = _useTranslation.i18n;
  return <div name={t(\\"7d3c35c2\\")}>{t(\\"df484e72\\")}</div>;
};"
`;

exports[`auto i18n plugin unit tests stringLiteral cases handle arrayExpression 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.Component = void 0;
var _reactI18next = require(\\"react-i18next\\");
var Component = exports.Component = function Component() {
  var _useTranslation = (0, _reactI18next.useTranslation)(),
    t = _useTranslation.t,
    i18n = _useTranslation.i18n;
  var obj1 = [t(\\"e86e487c\\"), t(\\"3c3e23ee\\")];
  return <div name={t(\\"7d3c35c2\\")}>{t(\\"e2fd33fa\\")}</div>;
};"
`;

exports[`auto i18n plugin unit tests stringLiteral cases handle binary expression rendering 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.Component = void 0;
var _reactI18next = require(\\"react-i18next\\");
var showText = true;
var Component = exports.Component = function Component() {
  var _useTranslation = (0, _reactI18next.useTranslation)(),
    t = _useTranslation.t,
    i18n = _useTranslation.i18n;
  return <div>
              {showText & t(\\"63e700c1\\")}
            </div>;
};"
`;

exports[`auto i18n plugin unit tests stringLiteral cases handle conditional jsx attribute 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.Component = void 0;
var _reactI18next = require(\\"react-i18next\\");
var Component = exports.Component = function Component() {
  var _useTranslation = (0, _reactI18next.useTranslation)(),
    t = _useTranslation.t,
    i18n = _useTranslation.i18n;
  console.log('输出值');
  var var1 = true;
  return <div name={var1 ? t(\\"d610f7ab\\") : t(\\"d41501c8\\")}>{t(\\"df484e72\\")}</div>;
};"
`;

exports[`auto i18n plugin unit tests stringLiteral cases handle conditional rendering 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.Component = void 0;
var _reactI18next = require(\\"react-i18next\\");
var showText = true;
var Component = exports.Component = function Component() {
  var _useTranslation = (0, _reactI18next.useTranslation)(),
    t = _useTranslation.t,
    i18n = _useTranslation.i18n;
  return <>
            <div>{showText && t(\\"63e700c1\\")}</div>
            <span>{showText ? t(\\"d610f7ab\\") : t(\\"d41501c8\\")}</span>
            </>;
};"
`;

exports[`auto i18n plugin unit tests stringLiteral cases handle jsx attribute 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.Component = void 0;
var _reactI18next = require(\\"react-i18next\\");
var Component = exports.Component = function Component() {
  var _useTranslation = (0, _reactI18next.useTranslation)(),
    t = _useTranslation.t,
    i18n = _useTranslation.i18n;
  console.log('输出值');
  return <div name={t(\\"7d3c35c2\\")}>{t(\\"df484e72\\")}</div>;
};"
`;

exports[`auto i18n plugin unit tests stringLiteral cases handle message object 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.Component = void 0;
var _reactI18next = require(\\"react-i18next\\");
var Component = exports.Component = function Component() {
  var _useTranslation = (0, _reactI18next.useTranslation)(),
    t = _useTranslation.t,
    i18n = _useTranslation.i18n;
  useEffect(function () {
    message.info({
      content: t(\\"d44d48e7\\")
    });
  }, [t]);
  return <div name={t(\\"7d3c35c2\\")}>{t(\\"e2fd33fa\\")}</div>;
};"
`;

exports[`auto i18n plugin unit tests stringLiteral cases handle modal object 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.Component = void 0;
var _reactI18next = require(\\"react-i18next\\");
var Component = exports.Component = function Component() {
  var _useTranslation = (0, _reactI18next.useTranslation)(),
    t = _useTranslation.t,
    i18n = _useTranslation.i18n;
  var cb = useCallback(function () {
    modal.info({
      content: <div>{t(\\"d44d48e7\\")}</div>,
      footer: <>
              <Space>
              <Button>{t(\\"4d0b4688\\")}</Button>
              <Button>{t(\\"b56d9ac6\\")}</Button>
              </Space>
              </>
    });
  }, [t]);
  return <div name={t(\\"7d3c35c2\\")} onClick={cb}>{t(\\"e2fd33fa\\")}</div>;
};"
`;

exports[`auto i18n plugin unit tests stringLiteral cases handle module scope objectProperty 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.Component = void 0;
var _reactI18next = require(\\"react-i18next\\");
var obj1 = {
  name: \\"属性值1\\",
  key: \\"属性值2\\"
};
var Component = exports.Component = function Component() {
  var _useTranslation = (0, _reactI18next.useTranslation)(),
    t = _useTranslation.t,
    i18n = _useTranslation.i18n;
  return <div name={t(\\"7d3c35c2\\")}>{t(\\"e2fd33fa\\")}</div>;
};"
`;

exports[`auto i18n plugin unit tests stringLiteral cases handle nested JSXText 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.Component = void 0;
var _reactI18next = require(\\"react-i18next\\");
var Component = exports.Component = function Component() {
  var _useTranslation = (0, _reactI18next.useTranslation)(),
    t = _useTranslation.t,
    i18n = _useTranslation.i18n;
  return <div>
              {t(\\"cd713656\\")}
              <div>{t(\\"7eaacb74\\")}</div>
            </div>;
};"
`;

exports[`auto i18n plugin unit tests stringLiteral cases handle objectProperty 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.Component = void 0;
var _reactI18next = require(\\"react-i18next\\");
var Component = exports.Component = function Component() {
  var _useTranslation = (0, _reactI18next.useTranslation)(),
    t = _useTranslation.t,
    i18n = _useTranslation.i18n;
  var obj1 = {
    name: t(\\"e86e487c\\"),
    key: t(\\"3c3e23ee\\")
  };
  return <div name={t(\\"7d3c35c2\\")}>{t(\\"e2fd33fa\\")}</div>;
};"
`;

exports[`auto i18n plugin unit tests stringLiteral cases handle string literals combined with variables 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.Component = void 0;
var _reactI18next = require(\\"react-i18next\\");
var text = \\"模块作用域文案\\";
var Component = exports.Component = function Component() {
  var _useTranslation = (0, _reactI18next.useTranslation)(),
    t = _useTranslation.t,
    i18n = _useTranslation.i18n;
  return <div>{text}{t(\\"c4899352\\")}</div>;
};"
`;

exports[`auto i18n plugin unit tests stringLiteral cases handle svg react component 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.Component = void 0;
var _reactI18next = require(\\"react-i18next\\");
var Component = exports.Component = function Component() {
  var _useTranslation = (0, _reactI18next.useTranslation)(),
    t = _useTranslation.t,
    i18n = _useTranslation.i18n;
  return <svg t=\\"1711425339799\\" className=\\"icon\\" viewBox=\\"0 0 1024 1024\\" version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"200\\" height=\\"200\\">
        <path id=\\"svg中的中文\\" d=\\"M448 672c-6.4 0-19.2 0-25.6-6.4-12.8-12.8-12.8-32 0-44.8L531.2 512 422.4 409.6c-12.8-12.8-12.8-32 0-44.8s32-12.8 44.8 0l128 128c12.8 12.8 12.8 32 0 44.8l-128 128C467.2 672 454.4 672 448 672z\\" fill=\\"#272636\\" />
      </svg>;
};"
`;

exports[`auto i18n plugin unit tests stringLiteral cases handle templateElement text 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.Component = void 0;
var _reactI18next = require(\\"react-i18next\\");
var Component = exports.Component = function Component() {
  var _useTranslation = (0, _reactI18next.useTranslation)(),
    t = _useTranslation.t,
    i18n = _useTranslation.i18n;
  return <div>
              {t(\\"cd713656\\")}
              <div>{t(\\"7eaacb74\\")}</div>
            </div>;
};"
`;
